sections/shared/Layout/AppLayout/Header/UserMenu/UserMenu.tsx

```diff
@@ -56,6 +59,23 @@ const UserMenu: FC<UserMenuProps> = ({ isTextButton }) => {
 		[redeemableDeprecatedSynths?.totalUSDBalance]
 	);
 
+	/**
+	 * @dev We require this new provider since we need one connected to Ethereum
+	 * mainnet and NOT to the Optimism network.
+	 */
+	let ensProvider = new ethers.providers.InfuraProvider();
```

Comment:
Looks like this works. Can we move this to the `connector.tsx` and expose `staticMainnetProvider` and then use that here like so: `const { connectWallet, signer, staticMainnetProvider } = Connector.useContainer();`?